synapse/storage/databases/main/media_repository.py

```diff
@@ -361,6 +406,30 @@ async def mark_local_media_as_safe(self, media_id: str, safe: bool = True) -> No
             desc="mark_local_media_as_safe",
         )
 
+    async def count_pending_media(self, user_id: UserID) -> Tuple[int, int]:
+        """Count the number of pending media for a user.
+        Returns:
+            A tuple of two integers: the total pending media requests and the earliest
+            expiration timestamp.
+        """
+
+        def get_pending_media_txn(txn: LoggingTransaction) -> Tuple[int, int]:
+            sql = (
+                "SELECT COUNT(*), MIN(unused_expires_at)"
+                " FROM local_media_repository"
+                " WHERE user_id = ?"
+                " AND quarantined_by IS NULL"
```

Comment:
Why do we care about excluding media that is quarantined?

This doesn't appear to be some efficient use of an index thing to get a smaller dataset to work with.

Reply:
I guess it really doesn't matter to exclude quarantined here. Probably better if we don't, since if someone has media quarantined, they probably are a malicious party and shouldn't be allowed to upload more media.